webapp/javascript/redux/reducers/settings.ts

```diff
@@ -34,15 +35,29 @@ type ApiKeysState = {
   data?: APIKeys;
 };
 const apiKeysInitialState: ApiKeysState = { type: 'pristine', data: undefined };
+let reloadApiKeysAbortController: AbortController | undefined;
 
 export const reloadApiKeys = createAsyncThunk(
   'newRoot/reloadAPIKeys',
   async (_, thunkAPI) => {
-    const res = await fetchAPIKeys();
+    if (reloadApiKeysAbortController) {
+      reloadApiKeysAbortController.abort();
+    }
+    reloadApiKeysAbortController = new AbortController();
+
+    thunkAPI.signal = reloadApiKeysAbortController.signal;
```

Comment:
Do we need this? The `thunkAPI` already has a AbortSignal (via `.signal`), can't we just make it so that the component `useEffect` cleanup will call `.abort`? That way we don't have to deal with this at the redux level.
Now that I am thinking it looks like we need this for the exact case you showed: clicking on a button multiple times. Nevermind!

Ugh this would be so much easier to solve with observables!

Reply:
> Now that I am thinking it looks like we need this for the exact case you showed: clicking on a button multiple times. Nevermind!
Yes, that was done to cancel both `fetch` and thunk